opa-manager-operator/src/charm.py

```diff
@@ -0,0 +1,220 @@
+#!/usr/bin/env python3
+import os
+import logging
+import yaml
+import utils
+from pathlib import Path
+from ops.charm import CharmBase
+from ops.main import main
+from ops.framework import StoredState
+from ops.model import ActiveStatus, MaintenanceStatus, BlockedStatus
+
+
+from charmhelpers.core.hookenv import (
+    log,
+)
+from jinja2 import Template
+
+
+logger = logging.getLogger(__name__)
+
+
+class CustomResourceDefintion(object):
+    def __init__(self, name, spec):
+
+        self._name = name
+        self._spec = spec
+
+    @property
+    def spec(self):
+        return self._spec
+
+    @property
+    def name(self):
+        return self._name
+
+
+class OPAManagerCharm(CharmBase):
+    """
+    A Juju Charm for OPA
+    """
+
+    _stored = StoredState()
+
+    def __init__(self, *args):
+        super().__init__(*args)
+        self.framework.observe(self.on.config_changed, self._on_config_changed)
+        self.framework.observe(self.on.update_status, self._on_update_status)
+        self.framework.observe(self.on.stop, self._on_stop)
+        self.framework.observe(self.on.install, self._on_install)
+        self.framework.observe(self.on.start, self._on_start)
+        self._stored.set_default(things=[])
+
+    def _on_config_changed(self, _):
+        """
+        Set a new Juju pod specification
+        """
+        self._configure_pod()
+
+    def _on_stop(self, _):
+        """
+        Mark unit is inactive
+        """
+        self.unit.status = MaintenanceStatus("Pod is terminating.")
+        logger.info("Pod is terminating.")
+
+    def _load_yaml_objects(self, files_list):
+        yaml_objects = []
+        try:
+            yaml_objects = [yaml.load(Path(f).read_text()) for f in files_list]
+        except yaml.YAMLError as exc:
+            print("Error in configuration file:", exc)
+
+        return yaml_objects
+
+    def _on_install(self, event):
+        logger.info("Congratulations, the charm was properly installed!")
+
+    def _on_update_status(self, event):
+        logger.info("Status updated")
```

Comment:
We didn't actually update any status. I would just remove this.

Reply:
removed